viamrobotics/prime
# APP-2072 Migrate PRIME to Svelte 4.

@mcous thanks for the inspiration for PR descriptions :)

## Overview

This migrates PRIME to Svelte 4. I wanted to do this migration before changing the package structure.

## Change log

- `npx svelte-migrate@latest svelte-4`
- Fix package version mismatches
- Import styles directly instead of `addStyles` helper
- Fix button and input `attachInternals`
- Fix code editor injected styles
- Change slider `range` prop to only accept a string - mixed types doesn't play well with web components
- Use Svelte `createEventDispatcher` directly - `get_current_component` no longer works
- Fix problematic tests

## Review requests

Pull down the branch and make sure:
- `pnpm run test` passes
- Elements look/behave correctly with `pnpm run storybook` and `pnpm start`

Questions:
- Is the minor version bump to 0.3.0 appropriate? I imagine we'll bump to 1.0.0 with the package restructure since that will be breaking.
